### ISSUE TITLE
"get_unknown" no longer compiles (Win 1809, latest VS and cppwinrt) 

### ISSUE THREAD

**Issue by ShawnKendall** [MicrosoftDocs/windows-uwp#1305]
**get_unknown** no longer compiles (Win 1809, latest VS and cppwinrt) as per example above and in the example Direct2D app.

```
check_hresult(factory-&gt;CreateSwapChainForCoreWindow(
    device.get(),
     get_unknown(CoreWindow::GetForCurrentThread()),
    &props,
    nullptr, // all or nothing
    swapChain.put())
);
```

`...direct2d\app.cpp(130): error C3861: 'get_unknown': identifier not found`

Already had the the command-line options mentioned elsewhere (as was working before updates anyway) :
/bigobj /await /permissive- /Zc:twoPhase-

Thanks for any help!

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2a2b07f2-2bb1-194b-2c23-5e3a2ce3635c
* Version Independent ID: 1c237c95-2767-3083-77bb-5d6e8e9064bb
* Content: [Move to C++/WinRT from WRL - UWP app developer](https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/move-to-winrt-from-wrl)
* Content Source: [windows-apps-src/cpp-and-winrt-apis/move-to-winrt-from-wrl.md](https://github.com/MicrosoftDocs/windows-uwp/blob/docs/windows-apps-src/cpp-and-winrt-apis/move-to-winrt-from-wrl.md)
* Product: **uwp**
* GitHub Login: @stevewhims
* Microsoft Alias: **stwhi**

**Comment by ShawnKendall** [comment]
Tried the updated call about but same result:

		agile_ref<winrt::Windows::UI::Core::CoreWindow> m_window;
		check_hresult(
		factory->CreateSwapChainForCoreWindow(
			device.get(),
			get_unknown(m_window.get()),
			&props,
			nullptr, // all or nothing
			swapChain.put())
		);

\direct2d\app.cpp(134): error C3861: 'get_unknown': identifier not found

**Comment by ginami** [comment]
Hi @ShawnKendall -- is this a duplicate of issue #1304 ? If you let me know the URL of the relevant doc, I can route your issue to the correct author. (This is a feedback channel for our docs.)

**Comment by ginami** [comment]
Thanks @ShawnKendall -- sorry, I see you did post this on the page...and you said you found a work-around, but let's see if @stevewhims can help.

**Comment by stevewhims** [comment]
Hi, @ShawnKendall! I'm just checking whether you're unblocked now? In the other issue (#1304), you say you've found a workaround. I've verified that the code in this topic that calls CreateSwapChainForCoreWindow, and uses winrt::get_unknown, is correct. I created a new CoreWindow project targeting the latest SDK (10.0.17763.0), added the appropriate headers and variables, and pasted in the code from the topic and it compiles. winrt::get_unknown is definitely still there. I also checked the [Full source code listing of a minimal Direct2D application](https://docs.microsoft.com/windows/uwp/cpp-and-winrt-apis/consume-com#full-source-code-listing-of-a-minimal-direct2d-application), and I found an issue with that, because it was using a macro (WINRT_TRACE) that's no longer in winrt/base.h. So I fixed that, and it should be live tomorrow.

Thanks for filing this issue. Please let me know if I can help futher.

~Steve

**Comment by ShawnKendall** [comment]
Yes I was building and using the Direct2D application as reference and that is where it wasn't working for me. 
Thanks for the fix and responses!

**Comment by stevewhims** [comment]
Great, so I'll go ahead and close this issue now. Thanks! ~Steve
